src/main/java/com/redhat/cloud/common/clowder/configsource/ClowderConfigSource.java

```diff
@@ -343,6 +427,90 @@ private String getHostPortDb(DatabaseConfig database) {
+                throw new IllegalStateException("Could not parse any certificate in the file");
+            }
+
+            KeyStore truststore = KeyStore.getInstance("PKCS12");
+            truststore.load(null);
+
+            for (int i = 0; i < certificates.size(); i++) {
+                truststore.setCertificateEntry("cert-" + i, certificates.get(i));
+            }
+
+            char[] password = buildPassword(base64Certs.get(0));
+            this.clowderEndpointPath = writeKeystore(truststore, password);
+            this.clowderEndpointPassword = new String(password);
+        } catch (IOException ioe) {
+            throw new IllegalStateException("Couldn't load the certificate, but we were requested a truststore", ioe);
+        } catch (KeyStoreException kse) {
+            throw new IllegalStateException("Couldn't load the keystore format PKCS12", kse);
+        } catch (NoSuchAlgorithmException | CertificateException ce) {
+            throw new IllegalStateException("Couldn't configure the keystore", ce);
+        }
+    }
+
+    static List<String> readCerts(String certString) {
+        return Arrays.stream(certString.split("-----BEGIN CERTIFICATE-----"))
+                .filter(s -> !s.isEmpty())
+                .map(s -> Arrays.stream(s
+                                .split("-----END CERTIFICATE-----"))
+                        .filter(s2 -> !s2.isEmpty())
+                        .findFirst()
+                        .orElseThrow(() -> new IllegalStateException("Invalid certificate found"))
+
+                )
+                .map(String::trim)
+                .map(s -> s.replaceAll("\n", ""))
+                .collect(Collectors.toList());
+    }
+    private List<byte[]> parsePemCert(List<String> base64Certs) {
+        return base64Certs
+                .stream()
+                .map(cert -> Base64.getDecoder().decode(cert))
+                .collect(Collectors.toList());
+    }
+
+    private X509Certificate buildx509Cert(byte[] cert) {
+        try {
+            CertificateFactory factory = CertificateFactory.getInstance("X.509");
+            return (X509Certificate) factory.generateCertificate(new ByteArrayInputStream(cert));
+        } catch (CertificateException certificateException) {
+            throw new IllegalStateException("Couldn't load the x509 certificate factory", certificateException);
+        }
+    }
+
+    private String writeKeystore(KeyStore keyStore, char[] password) {
+        try {
+            File file = createTempFile("truststore", ".trust");
+            keyStore.store(new FileOutputStream(file), password);
+            return file.getAbsolutePath();
+        } catch (IOException | KeyStoreException | NoSuchAlgorithmException | CertificateException e) {
+            throw new RuntimeException("Truststore creation failed", e);
+        }
+    }
+
+    private char[] buildPassword(String seed) {
+        int size = Math.min(33, seed.length());
```

Comment:
Right - will update with comments/constant - the 33 is just a random size to avoid having a fixed password like the usual "changeme" somewhere in the code